### PR TITLE
SNOW-741323  Remove side effect of import statements

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-741323

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Unset `snowflake.connector.paramstyle` in import path, set it by adding an optional connection parameter.

### CODE REVIEW

File: src/snowflake/snowpark/session.py
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:240]: nit: Should we put the comment before `if`?

```
# To be...
if...
```

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/_internal/server_connection.py
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/_internal/server_connection.py:50]: add back this comment?

**Review by sfc-gh-jdu — APPROVED** [review]

**Comment by sfc-gh-stan** [comment]
Oh Jianzhun's comment was not displayed until this page refreshed after I clicked the merge button 😅 , will add a comment in my other PR.

**Pull request merged**
